src/Stratis.Features.SystemContracts/DispatcherRegistry.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Stratis.Features.SystemContracts
+{
+    public interface IDispatcherRegistry
```

Comment:
Can the dispatch mechanism be implemented in such a way that any smart contract can "opt into" providing a dispatch implementation rather than making it mandatory?

Reply:
We could but that's a substantial change to smart contracts and outside the scope of this work.

Reply:
On the contrary, I am asking that we don't make changes to the current smart contracts in the regards mentioned.

Reply:
The change you're suggesting in #542 affects how contracts are able to be written on Cirrus and is significant enough that would require updating documentation, further testing etc. It's more changes, not less.

The dispatcher implementation looks large but it's just boilerplate to handle dynamic dispatch and it's easy to understand.

Reply:
Another distinction is that Cirrus APIs are public APIs. Other people use them and they have permanence. These APIs are internal and much easier for us to change in the future.